src/main/java/org/folio/linked/data/service/impl/BatchIndexServiceImpl.java

```diff
@@ -23,28 +31,34 @@
 @RequiredArgsConstructor
 public class BatchIndexServiceImpl implements BatchIndexService {
 
-  private final KafkaSearchSender kafkaSearchSender;
   private final EntityManager entityManager;
+  private final KafkaSearchMessageMapper<LinkedDataWork> searchBibliographicMessageMapper;
+  @Qualifier("bibliographicMessageProducer")
+  private final FolioMessageProducer<ResourceIndexEvent> bibliographicMessageProducer;
 
   @Override
   public BatchIndexResult index(Stream<Resource> resources) {
     var recordsIndexed = new AtomicInteger(0);
     var indexedIds = resources
       .map(resource -> handleResource(resource, recordsIndexed))
-      .filter(Optional::isPresent)
       .flatMap(Optional::stream)
       .collect(Collectors.toSet());
     return new BatchIndexResult(recordsIndexed.get(), indexedIds);
   }
 
   private Optional<Long> handleResource(Resource resource, AtomicInteger recordsIndexed) {
     try {
-      boolean indexed = kafkaSearchSender.sendMultipleWorksCreated(resource);
-      if (indexed) {
-        recordsIndexed.getAndIncrement();
-      } else {
-        logFailure(resource.getId());
-      }
+      //TODO mapper mustn't decide validation problems
```

Comment:
Are we planning to fix this as part of this PR? If not, can we please create tech debt ticket?

Reply:
yes. I would like to do it separately,
Mapper must be simpler